### DISCUSSION TITLE
It seems podman doesn't read "remote" in config by now

### DISCUSSION THREAD

**Discussion by hsupu** [containers/podman#12864] [Q&A] [answered]
I set my `containers.conf` as below but it didn't work.

```conf
[engine] 
remote = true
```

I believe that the `podman` should behave like `podman -r` by default if the remote flag is on.

This feature is [documented](https://www.mankier.com/5/containers.conf).

I found the code is [here](https://github.com/containers/podman/blob/eeb76db0fd0dc2a1115c2f2ad12c82eadbc62509/vendor/github.com/containers/common/pkg/config/config.go#L342)

```go
// EngineConfig contains configuration options used to set up a engine runtime
type EngineConfig struct {
	// Indicates whether the application should be running in Remote mode
	Remote bool `toml:"remote,omitempty"`
}
```

I saw there is another struct (that looked critical) contains a `Remote bool`. It's [here](https://github.com/containers/podman/blob/eeb76db0fd0dc2a1115c2f2ad12c82eadbc62509/pkg/domain/entities/engine.go#L45)

```go
// PodmanConfig combines the defaults and settings from the file system with the
// flags given in os.Args. Some runtime state is also stored here.
type PodmanConfig struct {
	*config.Config
	Remote         bool       // Connection to Podman API Service will use RESTful API
}
```

I found the working function in code is `IsRemote` which is defined [here](https://github.com/containers/podman/blob/0144f46ac5067196019225430847691502d74da7/cmd/podman/registry/remote.go).

This feature, read remote flag from config file, does it not work by design, or it's just under development so I maybe could try to fix it ?

Thanks!

**Comment by rhatdan** [comment, marked as answer]
It is supposed to work. Looks like a bug.

**Reply by hsupu** [reply]
Thank you for quick response. I saw the PR #12871, It's amazing the team could fix this so fast. Thank you all.

**Reply by rhatdan** [reply]
We aim to please.
